src/main/java/ee/sk/smartid/SmartIdClient.java

```diff
@@ -233,10 +231,22 @@ private void populateBuilderFields(SmartIdRequestBuilder builder) {
     builder.withRelyingPartyName(relyingPartyName);
   }
 
-  private SessionStatusPoller createSessionStatusPoller(SmartIdRestConnector connector) {
+  private SessionStatusPoller createSessionStatusPoller(SmartIdConnector connector) {
     SessionStatusPoller sessionStatusPoller = new SessionStatusPoller(connector);
     sessionStatusPoller.setPollingSleepTime(pollingSleepTimeUnit, pollingSleepTimeout);
     sessionStatusPoller.setResponseSocketOpenTime(sessionStatusResponseSocketOpenTimeUnit, sessionStatusResponseSocketOpenTimeValue);
     return sessionStatusPoller;
   }
+
+  public SmartIdConnector getSmartIdConnector() {
+    if (null == connector) {
+      // Fallback to REST connector when not initialised
+      return new SmartIdRestConnector(hostUrl, networkConnectionConfig);
```

Comment:
If getSmartIdConnector is called multiple times, then a new SmartIdRestConnector instance is created every time. It would be a good idea to set the connector after creating it.